docs/webpack-dev-server.config.js

```diff
@@ -18,7 +18,12 @@ const config = {
       //material-ui requires will be searched in src folder, not in node_modules
       'material-ui/lib': path.resolve(__dirname, '../src'),
       'material-ui': path.resolve(__dirname, '../src'),
+      'material-ui-docs': path.resolve(__dirname, 'src/app'),
     },
+    fallback: [
```

Comment:
Without this, the module path resolution fails.

We need to add extra module resolution paths to support the `examples/` components being outside of `docs`, as they only use the root `node_modules` folder by default.

I first attempted to fix this using `resolve.modulesDirectories`, however I ran into an annoying edge case.

There are two libraries in `docs/package.json` that have a dependency on `babylon`. However, they are different versions/builds of `babylon`. One of the builds is a pre-minified build which completely breaks the docs site and webpack also complains about it.

What happens is when a module in `src/Component/examples/` looks for `recast` via the proptype parser in the `docs/node_modules` path when using `resolve.moduleDirectories`, `recast` does **not** use the nested `node_modules` folder inside `docs/node_modules/recast/node_modules/babylon`. Since `docs/node_modules` is just an extra resolved path for anything outside of `docs`, it finds `babylon` in `docs/node_modules/babylon` and uses that even though it is the wrong version and a pre-minified build.

By using `fallback` instead, this issue is circumvented as it resorts to `docs/node_modules` as a last ditch resort rather than prioritising it over the local `node_modules` folder in `docs/node_modules/recast`.